Userland/Libraries/LibWeb/WebGL/OpenGLContext.cpp

```diff
@@ -0,0 +1,357 @@
+/*
+ * Copyright (c) 2024, Aliaksandr Kalenik <kalenik.aliaksandr@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/OwnPtr.h>
+#include <LibGfx/Bitmap.h>
+#include <LibWeb/WebGL/OpenGLContext.h>
+
+#ifdef HAS_ACCELERATED_GRAPHICS
+#    include <LibAccelGfx/Canvas.h>
+#    include <LibAccelGfx/Context.h>
+#elif defined(AK_OS_SERENITY)
+#    include <LibGL/GLContext.h>
+#endif
+
+namespace Web::WebGL {
+
+#ifdef HAS_ACCELERATED_GRAPHICS
+class AccelGfxContext : public OpenGLContext {
```

Comment:
It's unfortunate that we need the `AccelGfxContext` and `LibGLContext` separate classes, but mmmmkay.

Reply:
yes, it seems like something that totally could be done at compile time. although probably not even close to be perf problem at the moment so I decided it is going to be future project :sweat_smile: